SwrveSDKTest/src/test/java/com/swrve/sdk/utils/TestHttpServer.java

```diff
@@ -0,0 +1,99 @@
+package com.swrve.sdk.utils;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import fi.iki.elonen.NanoHTTPD;
+
+public class TestHttpServer extends NanoHTTPD {
```

Comment:
And you can then assert a method is called with a certain value by doing something like this:
https://github.com/Swrve/swrve-sdk/blob/release-4_6/android/SwrveSDKTests/src/test/java/com/swrve/sdk/LocationCampaignTest.java#L428